### PR TITLE
Cleaned up, moved more to user service from user controller

### PR DESCRIPTION
**Pull request by cwkingjr**

### CODE REVIEW

File: src/controllers/userController.js [hunk 1]
```diff
@@ -9,20 +9,13 @@ module.exports = {
 };
 
 function list(req, res, next) {
-	User
-		.findAll()
-		.then(userService.restrictOutputFields)
+	userService.getAllUsers()
```
Comment (charwking) [inline src/controllers/userController.js]: I think I like that the controller filters the output by calling `restrictOutputFields`. This allows anyone making calls to the `userService` to reasonably assume that returned values will be `User` objects, not generic objects that no longer have all the `sequelize` model instance methods.

File: src/services/userService.js [hunk 2]
```diff
@@ -59,20 +63,34 @@ function restrictOutputFields(users) {
...
-			.then(emailService.sendUserVerificationEmail)
-			.catch(function (err) {
-				// if the user already exists, no worries
-				if (!(err instanceof Sequelize.UniqueConstraintError)) {
-					throw err;
-				}
-			});
+		.findOrCreate({where: {email: email}})
+		.spread(function(user, created) {
+			if (created !== true) {
```
Comment (charwking) [inline src/services/userService.js]: `if (!created)`

File: src/controllers/userController.js (hunk 1 above)
Comment (cwkingjr) [inline src/controllers/userController.js]: Done

File: src/services/userService.js (hunk 2 above)
Comment (cwkingjr) [inline src/services/userService.js]: Done

**Comment by cwkingjr** [comment]
@charwking requested changes made.

**Comment by charwking** [comment]
I'd like to see the `getAllUsers` method removed, but don't care if that's this PR or a separate one. Everything else is okay.

**Pull request merged**